obsidian.el

```diff
@@ -69,110 +68,292 @@
+If it is true, a timer will be created using the values of
+`obsidian-cache-expiry' and `obsidian-update-idle-wait'."
+  :type 'boolean)
+
+(defcustom obsidian-cache-expiry (* 60 5)
+  "The number of seconds before the Obsidian cache will update."
+  :type 'integer
+  :group 'obsidian)
+
+(defcustom obsidian-update-idle-wait 5
+  "Seconds to wait after cache expiry for Emacs to be idle before running update."
+  :type 'integer
+  :group 'obsidian)
+
 (eval-when-compile (defvar local-minor-modes))
 
-(defun obsidian--directory-files-pre28 (orig-func dir &optional full match nosort ignored)
+(defun obsidian--directory-files-pre28
+    (orig-func dir &optional full match nosort _)
   "Version of `directory-files' compatible with Emacs versions < 28.
 
 ORIG-FUNC is the original `directory-files' function that is going to be
 advised,and DIR and the directory of files on which `directory-files' will
 be called.
 FULL, MATCH, and NOSORT are the optional arguments for the `directory-files'
-function, while IGNORED is the optional 4th argument used with newer versions
+function, while _ is the optional 4th argument used with newer versions
 of `dirctory-files'."
   (apply orig-func dir full match nosort))
 
 (if (< emacs-major-version 28)
     (advice-add 'directory-files :around #'obsidian--directory-files-pre28))
 
-;;;###autoload
 (defun obsidian-specify-path (&optional path)
-  "Specifies obsidian folder PATH to obsidian-folder variable.
+  "Set `obsidian-directory' to PATH or user-selected directory.
 
-When run interactively asks user to specify the path."
-  (interactive)
-  (->> (or path (read-directory-name "Specify path to Obsidian folder"))
+You most likely want to run `obsidian-change-vault'."
+  (->> (or path (read-directory-name "Specify path to Obsidian vault: "))
        (expand-file-name)
        (customize-set-value 'obsidian-directory)))
-(defvar obsidian--tags-list nil "List of Obsidian Notes tags generated by obsidian.el.")
 
-(defvar obsidian--tag-regex "#[[:alnum:]-_/+]+" "Regex pattern used to find tags in Obsidian files.")
+;;;###autoload
+(defun obsidian-change-vault (&optional path)
+  "Set vault directory to PATH and repopulate vault cache.
+
+When run interactively asks user to specify the path."
+  (interactive)
+  (message "Obsidian vault set to: %s" (obsidian-specify-path path))
+  (obsidian-clear-cache)
+  (obsidian-populate-cache))
+
+(define-minor-mode obsidian-mode
+  "Toggle minor `obsidian-mode' on and off.
+
+Interactively with no argument, this command toggles the mode.
+A positive prefix argument enables the mode, any other prefix
+argument disables it.  From Lisp, argument omitted or nil enables
+the mode, `toggle' toggles the state."
+  ;; The initial value.
+  :init-value nil
+  :lighter " obs"
+  :after-hook (obsidian-update)
+  :keymap (make-sparse-keymap))
+
+(defvar obsidian--debug-messages nil "Additional messages will be displayed for debugging.")
+
+(defvar obsidian--tag-regex

```

Comment:
The tag regex was updated to better align with what Obsidian Notes recognizes as valid tags.